src/mongo/MongoScrapbook.ts

```diff
@@ -120,11 +121,25 @@ async function executeCommand(activeEditor: vscode.TextEditor, database: MongoDa
 				await editorManager.showDocument(new MongoFindOneResultEditor(database, command.collection, result), 'cosmos-result.json', { showInNextColumn: true });
 			} else {
 				await vscodeUtil.showNewFile(result, extensionPath, 'result', '.json', activeEditor.viewColumn + 1);
+				await refreshTree();
 			}
 		}
 	} else {
 		throw new Error('No MongoDB command found at the current cursor location.');
 	}
+
+	async function refreshTree() {
```

Comment:
We try to avoid classes or functions within other functions. It can lead to weird scoping issues if you're not careful. Either move this to the top level in this file (and pass in the database) or just don't use a function for this since it's only called once

Reply:
I moved it to top level, I find executeCommand easier to read that way.